### PR TITLE
Fix warnings about raw types

### PR DESCRIPTION
**Pull request by proski**
```
*  Fix warnings about raw types
   
   Warnings caused by raw types in the Jenkins API are not fixed, as there
   is no way to do it.
   
   Remove casts that are redundant with parameterized types.
```

This would conflict with other PRs, but I'm ready to rebase them quickly if this PR is merged. Let's take care of the warnings and set a higher standard for the new code.

### CODE REVIEW

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java [hunk 1]
```diff
@@ -70,7 +70,8 @@ public StashPostBuildComment newInstance(StaplerRequest req, JSONObject formData
     }
 
     @Override
-    public boolean isApplicable(Class<? extends AbstractProject> jobType) {
+    public boolean isApplicable(
+        @SuppressWarnings("rawtypes") Class<? extends AbstractProject> jobType) {
```
Comment (jakub-bochenski) [inline src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java]: why is this needed?

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java [hunk 2]
```diff
@@ -14,11 +14,13 @@
 public class StashBuildEnvironmentContributor extends EnvironmentContributor {
   @Override
   public void buildEnvironmentFor(
-      @Nonnull Run r, @Nonnull EnvVars envs, @Nonnull TaskListener listener)
+      @SuppressWarnings("rawtypes") @Nonnull Run r,
```
Comment (jakub-bochenski) [inline src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java]: Does this come from the super definition and cannot be changed? I'm surprised we have to suppress it in that case

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java (hunk 1 above)
Comment (jakub-bochenski) [inline src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java]: same thing here, if this comes from super definition, why do we need to suppress it?

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java (hunk 2 above)
Comment (proski) [reply src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java]: Yes, it comes from the superclass:
https://javadoc.jenkins.io/hudson/model/EnvironmentContributor.html#buildEnvironmentFor-hudson.model.Run-hudson.EnvVars-hudson.model.TaskListener-

It causes a warning with Java 8. it's not from FindBugs. Eclipse marks it as "java warning".

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java (hunk 1 above)
Comment (proski) [reply src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java]: Yes, it comes from the superclass:
https://javadoc.jenkins-ci.org/hudson/tasks/BuildStepDescriptor.html#isApplicable-java.lang.Class-

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java (hunk 2 above)
Comment (jakub-bochenski) [reply src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java]: I'm pretty sure that Eclipse has an option to ignore warnings caused by super-type signatures.
Does javac print out a warning?

Comment (proski) [reply src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java]: You are right, there is an option in Eclipse, it can be found from the tooltip on the warning by selecting "Configure problem severity". It's called "Ignore unavoidable generic type problems due to raw APIs".

The compiler will issue warnings if #50 is applied:

```
[WARNING] /home/roskinp/src/jenkins/plugins/stash-pullrequest-builder-plugin/src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java:[73,49] found raw type: hudson.model.AbstractProject
  missing type arguments for generic class hudson.model.AbstractProject<P,R>
[WARNING] /home/roskinp/src/jenkins/plugins/stash-pullrequest-builder-plugin/src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java:[17,16] found raw type: hudson.model.Run
  missing type arguments for generic class hudson.model.Run<JobT,RunT>
```

But since we are not doing it, I think those suppressions can be removed. I think the Eclipse default settings are not good in that regard.

I'm glad I checked the warnings, it turns out the cast to `StashCause` in `StashBuildEnvironmentContributor.java` is redundant now.

**Comment by proski** [comment]
Removed all suppressions. Removed another redundant cast.

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java (hunk 1 above)
Comment (proski) [reply src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java]: Eclipse doesn't actually suppress this warning, but I can live with it. It's likely an Eclipse bug.

**Review by jakub-bochenski — APPROVED** [review]
